packages/core/src/index.ts

```diff
@@ -227,20 +227,59 @@ function init(options: InitOptions): OnboardAPI {
   theme && updateTheme(theme)
 
   // handle auto connection of last wallet
-  if (connect && connect.autoConnectLastWallet) {
-    const lastConnectedWallet = getLocalStore(
-      STORAGE_KEYS.LAST_CONNECTED_WALLET
+  if (
+    connect &&
+    (connect.autoConnectLastWallet || connect.autoConnectAllPreviousWallet)
+  ) {
+    const lastConnectedWallets = JSON.parse(
+      getLocalStore(STORAGE_KEYS.LAST_CONNECTED_WALLET)
     )
 
-    lastConnectedWallet &&
+    // Handle for legacy single wallet approach
+    if (lastConnectedWallets && typeof lastConnectedWallets === 'string') {
       API.connectWallet({
-        autoSelect: { label: lastConnectedWallet, disableModals: true }
+        autoSelect: { label: lastConnectedWallets, disableModals: true }
       })
+    }
+    if (lastConnectedWallets && Array.isArray(lastConnectedWallets)) {
```

Comment:
```suggestion
    if (lastConnectedWallets && Array.isArray(lastConnectedWallets && lastConnectedWallets.length)) {
```

Do we need to ensure there are wallets in the array here?

Reply:
Ahh good call! Added!!